code/obj/item/kitchen.dm

```diff
@@ -568,27 +571,32 @@ TRAYS
 	/// Attempts to add an item to the plate, if there's space. Returns TRUE if food is successfully added.
 	proc/add_contents(obj/item/food, mob/user, click_params)
 		. = FALSE
-		if (istype(food, /obj/item/plate))
+		if (istype(food, /obj/item/plate) && src.is_plate)
 			if (food == src)
-				boutput(user, "<span class='alert>You can't stack a plate on itself!</span>")
+				boutput(user, "<span class='alert'>You can't stack a [src] on itself!</span>")
 				return
 			var/obj/item/plate/not_really_food = food
 			. = src.stackable && not_really_food.stackable // . is TRUE if we can stack the other plate on this plate, FALSE otherwise
 
-		if (length(src.contents) == max_food)
+		if (length(src.foods_inside) == max_food && src.is_plate)
 			boutput(user, "<span class='alert'>There's no more space on \the [src]!</span>")
 			return
-		if (!food.edible && !.) // plates aren't edible, so we check if we're adding a valid plate as well (. is TRUE if so)
+			                                    // anything that isn't a plate may as well hold anything that fits the plate
+		if (!food.edible && !. && src.is_plate) // plates aren't edible, so we check if we're adding a valid plate as well (. is TRUE if so)
 			boutput(user, "<span class='alert'>That's not food, it doesn't belong on \the [src]!</span>")
 			return
-		if (food.w_class > W_CLASS_NORMAL)
-			boutput(user, "You try to think of a way to put [food] on \the [src] but it's not possible! It's too large!")
+		if (food.w_class > W_CLASS_NORMAL && !istype(food, /obj/item/plate/))
```

Comment:
😩

Reply:
i changed it to something prettier. i'll leave this review open until you say it's fine though.

Reply:
no i really liked it lol, just being sad over infinite plate stacking 😿